tests/test-softhsm.sh

```diff
@@ -0,0 +1,171 @@
+    pkcs11-tool --keypairgen --key-type="RSA:2048" --login --pin=$PINVALUE --module="$P11LIB" --label="RSA" --id="$KEYID"
+
+    BASEURIWITHPIN="pkcs11:id=${URIKEYID};pin-value=${PINVALUE}"
+    BASEURI="pkcs11:id=${URIKEYID}"
+    PUBURI="pkcs11:type=public;id=${URIKEYID}"
+    PRIURI="pkcs11:type=private;id=${URIKEYID}"
+
+    title LINE "RSA PKCS11 URIS"
+    echo "${BASEURIWITHPIN}"
+    echo "${BASEURI}"
+    echo "${PUBURI}"
+    echo "${PRIURI}"
+    echo ""
+
+    # generate ECC key pair
+    KEYID='0002'
+    URIKEYID="%00%02"
+    pkcs11-tool --keypairgen --key-type="EC:secp256r1" --login --pin=$PINVALUE --module="$P11LIB" --label="ECC" --id="$KEYID"
+
+    ECBASEURIWITHPIN="pkcs11:id=${URIKEYID};pin-value=${PINVALUE}"
+    ECBASEURI="pkcs11:id=${URIKEYID}"
+    ECPUBURI="pkcs11:type=public;id=${URIKEYID}"
+    ECPRIURI="pkcs11:type=private;id=${URIKEYID}"
+
+    KEYID='0003'
+    URIKEYID="%00%03"
+    pkcs11-tool --keypairgen --key-type="EC:secp256r1" --login --pin=$PINVALUE --module="$P11LIB" --label="PeerECC" --id="$KEYID"
+
+    ECPEERBASEURIWITHPIN="pkcs11:id=${URIKEYID};pin-value=${PINVALUE}"
+    ECPEERBASEURI="pkcs11:id=${URIKEYID}"
+    ECPEERPUBURI="pkcs11:type=public;id=${URIKEYID}"
+    ECPEERPRIURI="pkcs11:type=private;id=${URIKEYID}"
+
+    title LINE "EC PKCS11 URIS"
+    echo "${ECBASEURIWITHPIN}"
+    echo "${ECBASEURI}"
+    echo "${ECPUBURI}"
+    echo "${ECPRIURI}"
+    echo "${ECPEERBASEURIWITHPIN}"
+    echo "${ECPEERBASEURI}"
+    echo "${ECPEERPUBURI}"
+    echo "${ECPEERPRIURI}"
+    echo ""
+
+    title PARA "Show contents of softhsm token"
+    echo " ----------------------------------------------------------------------------------------------------"
+    pkcs11-tool -O --login --pin=$PINVALUE --module="$P11LIB"
+    echo " ----------------------------------------------------------------------------------------------------"
+
+    title PARA "Start the p11-kit server and check if it works"
+    echo " ----------------------------------------------------------------------------------------------------"
+    # p11-kit complains if there is not runtime directory
+    if [ -z "$XDG_RUNTIME_DIR" ]; then
+        export XDG_RUNTIME_DIR=$PWD
+    fi
+    eval $(p11-kit server --provider "$P11LIB" "pkcs11:")
+    echo " ----------------------------------------------------------------------------------------------------"
+    pkcs11-tool -O --login --pin=$PINVALUE --module="$P11KITCLIENTPATH"
+    echo " ----------------------------------------------------------------------------------------------------"
+
+    export PKCS11_PROVIDER_MODULE=$P11KITCLIENTPATH
+    #export PKCS11_PROVIDER_MODULE=/usr/lib64/pkcs11-spy.so
+    #export PKCS11SPY=$P11KITCLIENTPATH
+    export NOPSS="1"
+    export NOECCSHA2="1"
+    export NOOAEPSHA2="1"
+    export NODERIVE="1"
+
+
+    title LINE "Export variables to ${TMPPDIR}/debugvars for easy debugging"
+#TODO
+    BASEDIR=$(pwd)
+    cat > ${TMPPDIR}/debugvars <<DBGSCRIPT
```

Comment:
probably better to have debugvars.softhsm ?
(and debugvars.softokn on the other file)

Reply:
certainly good idea if they will run at the same time

Reply:
I think it is sufficient to use different tmp-dirs. I don't see any benefit to also use a module-specific ending for the debugvars file (or any other file).